tstore/tslong/tslong.py

```diff
@@ -1,40 +1,135 @@
 """Module defining the TSLong abstract wrapper."""
 
-from tstore.backend import DaskDataFrame, DataFrame, PandasDataFrame, PolarsDataFrame, PyArrowDataFrame
+from abc import abstractmethod
+from typing import TYPE_CHECKING, Optional
+
+import polars as pl
+import pyarrow as pa
+
+from tstore.backend import (
+    Backend,
+    DaskDataFrame,
+    DataFrame,
+    PandasDataFrame,
+    PolarsDataFrame,
+    PyArrowDataFrame,
+    change_backend,
+    re_set_dataframe_index,
+)
 from tstore.tswrapper.tswrapper import TSWrapper
 
+if TYPE_CHECKING:
+    # To avoid circular imports
+    from tstore.tsdf.tsdf import TSDF
+    from tstore.tswide.tswide import TSWide
+
 
 class TSLong(TSWrapper):
-    """Abstract wrapper for a long-form timeseries dataframe."""
+    """Abstract wrapper for a long-form timeseries DataFrame."""
+
+    def __init__(
+        self,
+        df: DataFrame,
+        id_var: str,
+        time_var: str = "time",
+        ts_vars: Optional[dict[str, list[str]]] = None,
+        static_vars: Optional[list[str]] = None,
+    ) -> None:
+        """Wrap a long-form timeseries DataFrame as a TSLong object.
+
+        Args:
+            df (DataFrame): DataFrame to wrap.
+            id_var (str): Name of the column containing the identifier variable.
+            time_var (str): Name of the column containing the time variable. Defaults to "time".
+            ts_vars (dict[str, list[str]]): Dictionary of named groups of column names.
+                Defaults to None, which will group all columns not in `static_vars` together.
+            static_vars (list[str]): List of column names that are static across time. Defaults to None.
+        """
+        if isinstance(df, (DaskDataFrame, PandasDataFrame)):
+            df[id_var] = df[id_var].astype("large_string[pyarrow]")
+
+        elif isinstance(df, PolarsDataFrame):
+            df = df.cast({id_var: pl.String})
+
+        elif isinstance(df, PyArrowDataFrame):
+            schema = df.schema
+            field_index = schema.get_field_index(id_var)
+            schema = schema.remove(field_index)
+            schema = schema.insert(field_index, pa.field(id_var, pa.large_string()))
+            df = df.cast(target_schema=schema)
+
```

Comment:
I would add here the renaming of the custom timesteps columns to "time" if differently specified ....